### ISSUE TITLE
[BUG] Fuzzing integration build (Linux) - invalid argument '-fsanitize=address' not allowed with '-fsanitize=thread'

### ISSUE THREAD

**Issue by zcduthie** [project-chip/connectedhomeip#25959]
### Reproduction steps

Following the libfuzzer command for linux listed under 'Fuzzing integration' instructions [here](https://github.com/project-chip/connectedhomeip/tree/master/examples/all-clusters-app/linux):
```./scripts/run_in_build_env.sh "./scripts/build/build_examples.py --target linux-x64-all-clusters-no-ble-asan-libfuzzer-clang build"```

On a fresh linux environment (Docker container on Windows host) results in the error:

```
clang-15: error: invalid argument '-fsanitize=address' not allowed with '-fsanitize=thread'
```

### Bug prevalence

Whenever I do this

### GitHub hash of the SDK that was being used

d81d4e3

### Platform

windows

### Anything else?

I believe this is because the following line enables tsan (while the sample command manually includes `-asan-`):
https://github.com/project-chip/connectedhomeip/blob/d81d4e367d4ba80eef751675ce29ebfdbba74c84/scripts/build/build/targets.py#L143

Additionally, if I bypass this manually (by updating targets.py to also default to asan) it seems to only produce a standard (non-libfuzzer) binary.
```
ubuntu@99a0803337d7:~/connectedhomeip$ ls out/linux-x64-all-clusters-no-ble-asan-libfuzzer-clang/
args.gn        build.ninja.stamp          compile_commands.json  liblogging-stdio.a  relative_path_transformations.json
build.ninja    chip-all-clusters-app      gen                    obj                 toolchain.ninja
build.ninja.d  chip-all-clusters-app.map  lib                    python              zap_gen.lock
```
(I believe the output binary should be `chip-all-clusters-app-fuzzing`). Executing the binary results in standard all-clusters-app operation, with no libfuzzer output:
```
ubuntu@99a0803337d7:~/connectedhomeip$ ./out/linux-x64-all-clusters-no-ble-asan-libfuzzer-clang/chip-all-clusters-app
CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /tmp/chip_kvs
CHIP:DL: writing settings to file (/tmp/chip_kvs-cwM2BF)
CHIP:DL: renamed tmp file to file (/tmp/chip_kvs)
CHIP:DL: ChipLinuxStorage::Init: Using KVS config file: /tmp/chip_kvs
CHIP:DL: ChipLinuxStorage::Init: Attempt to re-initialize with KVS config file: /tmp/chip_kvs
```

**Comment by bzbarsky-apple** [comment]
@andy31415

**Comment by andy31415** [comment]
it looks to me like `linux-x64-all-clusters-no-ble-asan-libfuzzer-clang` is not a valid target, however `linux-x64-all-clusters-no-ble-libfuzzer-clang` works for me.

we should probably update our documentation.

**Comment by andy31415** [comment]
oh .. typo in variant ... should not be asan, should be fuzzer. Will create a PR.

**Comment by andy31415** [comment]
Fixed ... als tried it to see it runs ... it does and crashes right away. We may need to fix some things ...

**Comment by andy31415** [comment]
```
CHIP:SPT: VerifyOrDie failure at ../../examples/all-clusters-app/linux/third_party/connectedhomeip/src/platform/CommissionableDataProvider.cpp:32: gCommissionableDataProvider != nullptr

```

It seems fuzzing was not run for a long time, so actual prerequisites are not set in main.

**Comment by andy31415** [comment]
DataProvider fixed in #25968
